backtrader/feeds/psqlfeed

```diff
@@ -0,0 +1,101 @@
+# !/usr/bin/env python
+# -*- coding: utf-8; py-indent-offset:4 -*-
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from backtrader import DataBase, date2num
+import datetime as dt
+import psycopg2
+from urllib.parse import urlparse
+
+
+# Used for fetching prices, adhering to the backtest library Database API
+
+
+class PsqlDatabase(DataBase):
+    '''
+    Fetches data from PSQL, and wraps it into a Feed consumable by cerebro
+    takes url connection string in form of :
+        postgres://{user}:{pw}@{host}:{port [optional]}/{database}
+
+    assumes a table name conforming to a schema similar to the following:
+            ticker        char(5),
+            date          date,
+            high          numeric(10,4),
+            low           numeric(10,4),
+            open          numeric(10,4),
+            close         numeric(10,4),
+            volume        integer,
+            unique (ticker, date)
+    '''
+
+    params = (
+        ('url', None),
+        ('dataname', None),
+        ('table', None),
+        ('fromdate', None),
+        ('todate', None),
+        ('ticker', None),
+    )
+
+    def start(self):
+        try:
+            conn = self._connect_db()
+            cursor = conn.cursor()
+        except Exception as e:
+            print('Failed to establish connection to PSQL: {}'.format(e))
+            return
+
+        query = ("""SELECT date, high, low, open, close, volume """
+                 """FROM {table} """
+                 """WHERE ticker = '{ticker}' """
+                 .format(table=self.p.table,
+                         ticker=self.p.ticker))
+
+        if self.p.fromdate is not None:
+            query += " AND date >= '{fromdate}' ".format(fromdate=dt.datetime.strftime(self.p.fromdate, '%Y-%m-%d'))
+        if self.p.todate is not None:
+            query += " AND date <= '{todate}' ".format(todate=dt.datetime.strftime(self.p.fromdate, '%Y-%m-%d'))
+
+        query += """ORDER BY date asc"""
+
+        try:
+            cursor.execute(query)
+        except Exception as e:
+            print('PSQL query failed: {}'.format(e))
+
+        self.price_rows = cursor.fetchall()
+        conn.close()
+
+        self.price_i = 0
+        super(PsqlDatabase, self).start()
+
+    def _load(self):
+        if self.price_i >= len(self.price_rows):
+            return False
+        row = self.price_rows[self.price_i]
+        self.price_i += 1
+
+        self.lines.datetime[0] = date2num(row[0])
```

Comment:
Suggest using a map from column to column index ( see GenericCSVData for example)  instead of using raw index constants. This way inheriting this class (in case of different db scheme) will be easier and will probably not require rewriting the `_load` method

Reply:
this is a neat trick, implemented.